### PR TITLE
Some doc and scala style fixes. Also adopted Spark's scala style.

### PR DESCRIPTION
**Pull request by fastier-li**

### CODE REVIEW

File: photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala [hunk 1]
```diff
@@ -403,19 +404,23 @@ object AvroDataReader {
...
-      case FloatType => checkNull(record, field).orElse(Some(Utils.getFloatAvro(record, field.name)))
-      case LongType => checkNull(record, field).orElse(Some(Utils.getLongAvro(record, field.name)))
-      case MapType(_, _, _) => Some(Utils.getMapAvro(record, field.name, field.nullable))
-      case _ =>
-        // Unsupported field type. Drop this for now.
-        None
-    }
+  protected[data] def readColumnValuesFromRecord(
+      record: GenericRecord,
+      schemaFields: Seq[StructField]): Seq[Any] =
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala]: This should fit on one line.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala [hunk 2]
```diff
@@ -22,24 +22,24 @@ import java.io.{File => JFile}
...
+ *
+ * 1. One DB storage is partitioned into multiple pieces we call partitions. It should be generated and controlled by
+ *    [[com.linkedin.photon.ml.FeatureIndexingJob]]. The partition strategy is via the hashcode of the feature names,
+ *    following the rules defined in [[org.apache.spark.HashPartitioner]].
+ *
+ * 2. Each time a user queries the index of a certain feature, the index map will first compute the feature hashcode,
+ *    and then compute the expected partition for this hashcode.
+ *
+ * 3. Because the index partitions are built in parallel, without sharing information between them, each partition's
+ *    internal index always starts from 0. We keep an array of offsets to properly compute the indexes (keep them
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: "... compute the *indices* (keep ..."

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala [hunk 3]
```diff
@@ -57,14 +57,14 @@ class PalDBIndexMap extends IndexMap {
...
-    * @param isLocal default: false, if set false will use SparkFiles to access cached files; otherwise,
-    *                it will directly read from local files
-    * @return
-    */
+   * Load a storage at a particular path
+   *
+   * @param storePath     The directory where the storage is put
+   * @param partitionsNum The number of partitions, the storage contains
+   * @param isLocal       default: false, if set false will use SparkFiles to access cached files; otherwise,
+   *                      it will directly read from local files
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: We don't the consistent spacing thing right now. Whether we should or not is another question, but let's be consistent for now.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala [hunk 4]
```diff
@@ -57,14 +57,14 @@ class PalDBIndexMap extends IndexMap {
...
+   * @return
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: Missing description of what's returned.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala [hunk 5]
```diff
@@ -15,30 +15,41 @@
...
 import org.apache.hadoop.fs.Path
 import org.apache.spark.SparkContext
 
 
 /**
-  * A PalDBIndexMap loader
-  *
-  */
+ * A loader for IndexMaps stored in PalDB.
+ *
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala]: Extra whitespace can be removed.

File: photon-ml/README.md [hunk 6]
```diff
@@ -3,4 +3,58 @@
...
+                    │   └── game
+                    ├── projector
+                    ├── sampler
+                    ├── stat
+                    ├── supervised
+                    │   ├── classification
+                    │   ├── model
+                    │   └── regression
+                    └── util
+</code></pre>
```
Comment (fastier-li) [inline photon-ml/README.md]: Removed.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala (hunk 1 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala]: It does.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala (hunk 2 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: Not needed: http://grammarist.com/usage/indexes-indices/.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala (hunk 3 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: OK.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala (hunk 4 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: Fixed.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala (hunk 5 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMapLoader.scala]: Done.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala (hunk 2 above)
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: "Meanwhile, indices is generally preferred in mathematical, financial, and **technical contexts**, while indexes is relatively common in general usage."

However, I'll drop this out of respect for the craft of your counter-pedantry.

**Comment by fastier-li** [comment]
Well - Josh doesn't want to.

Thanks,

Frank

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala (hunk 2 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/PalDBIndexMap.scala]: Yeah, but it says in North-America and Canada, "indexes" is preferred. So if you do technical stuff in Canada, what should it be? Anyway. I could put ind-(exes/ices) to not ruffle any feathers at all?

**Pull request merged**